### PR TITLE
Fix relocation for py-tensorflow

### PR DESCRIPTION
**Pull request by iarspider**
Use a simpler solution suggested by Shahzad: https://github.com/cms-sw/cmsdist/pull/8644#issuecomment-1685968060

### CODE REVIEW

**Comment by iarspider** [comment]
please test

File: pip/tensorflow.file [hunk 1]
```diff
@@ -11,6 +16,12 @@ Requires: llvm
 %define source0 none
 %define source_file none
 
+%define PipPostInstall \
+  for f in %{RelocateFile} ; do \
+    [ ! -f %{i}/$f ] || sed -i -e 's|/[^ ]/%{cmsplatf}/external/|%{cmsroot}/%{cmsplatf}/external/|' %{i}/$f; \
```
Comment (smuzaffar) [inline pip/tensorflow.file]: oops, it should be
```
's|/[^ ]*/%{cmsplatf}/external/|%{cmsroot}/%{cmsplatf}/external/|'
```
i.e. add `*` to match all not `space` chars

File: pip/tensorflow.file [hunk 2]
```diff
@@ -11,6 +16,12 @@ Requires: llvm
...
+  done
+
 %define PipPostPost \
-    %{relocateConfig}lib/python*/site-packages/tensorflow/include/external/local_config_cuda/cuda/cuda/cuda_config.h ; \
-    %{relocateConfig}lib/python*/site-packages/tensorflow/python/platform/build_info.py
+  for f in %{RelocateFile} ; do \
+    [ ! -f $RPM_INSTALL_PREFIX/$f ] || %{relocateConfig}/$f
```
Comment (smuzaffar) [inline pip/tensorflow.file]: missing `\` at the end here

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test

File: pip/tensorflow.file [hunk 3]
```diff
@@ -17,11 +17,11 @@ Requires: llvm
...
-    [ ! -f %{i}/$f ] || sed -i -e 's|/[^ ]*/%{cmsplatf}/external/|%{cmsroot}/%{cmsplatf}/external/|' %{i}/$f; \
+  for f in %{RelocateFile} ; do \\\
+    [ ! -f %{i}/$f ] || sed -i -e 's|/[^ ]*/%{cmsplatf}/external/|%{cmsroot}/%{cmsplatf}/external/|' %{i}/$f; \\\
   done
 
 %define PipPostPost \
-  for f in %{RelocateFile} ; do \
-    [ ! -f $RPM_INSTALL_PREFIX/$f ] || %{relocateConfig}/$f \
+  for f in %{RelocateFile} ; do \\\
+    [ ! -f $RPM_INSTALL_PREFIX/$f ] || %{relocateConfig}/$f \\\
```
Comment (smuzaffar) [inline pip/tensorflow.file]: I think you were just missing `;` here

**Comment by iarspider** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals

**Pull request merged**